sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -21,21 +29,141 @@ namespace Azure { namespace Core { namespace Test {
 
+  private:
+    // Call this method to update client options with the required configuration to
+    // support Record & Playback.
+    // If Playback or Record is not set, no changes will be done to the clientOptions or credential.
+    // Call this before creating the SDK client
+    void PrepareClientOptions(
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        Azure::Core::_internal::ClientOptions& options)
+    {
+      // Set up client options depending on the test-mode
+      if (m_testContext.IsPlaybackMode())
+      {
+        // Playback mode uses:
+        //  - playback transport adapter to read and return payload from json files
+        //  - never-expiring test credential to never require a token
+        options.Transport.Transport = m_interceptor->GetPlaybackClient();
+        credential = m_interceptor->GetTestCredential();
+      }
+      else if (!m_testContext.IsLiveMode())
+      {
+        // Record mode uses:
+        //  - curl or winhttp transport adapter
+        //  - Recording policy. Intercept server responses to create json files
+        // AZURE_TEST_RECORDING_DIR is exported by CMAKE
+        options.PerRetryPolicies.push_back(m_interceptor->GetRecordPolicy());
+      }
+    }
+
+    std::string Sanitize(std::string const& src)
+    {
+      std::string updated(src);
+      std::replace(updated.begin(), updated.end(), '/', '-');
+      return updated;
+    }
+
   protected:
     Azure::Core::Test::TestContextManager m_testContext;
     std::unique_ptr<Azure::Core::Test::InterceptorManager> m_interceptor;
 
+    // Reads the current test instance name.
+    // Name gets also sanitized (special chars are removed) to avoid issues when recording or
+    // creating
+    std::string GetTestName(bool sanitize = false)
+    {
+      std::string testName(::testing::UnitTest::GetInstance()->current_test_info()->name());
+      if (sanitize)
+      {
+        // replace `/` for `-`. Parameterized tests adds this char automatically to join the test
+        // name and the parameter suffix.
+        return Sanitize(testName);
+      }
+
+      return testName;
+    }
+
+    // Creates the sdk client for testing.
+    // The client will be set for record and playback before it is created.
+    template <class T, class O>
+    std::unique_ptr<T> InitTestClient(
+        std::string const& url,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        O& options)
+    {
+      // Run instrumentation before creating the client
+      PrepareClientOptions(credential, options);
+      return std::make_unique<T>(url, credential, options);
+    }
+
+    // Updates the time when test is on playback
+    void UpdateWaitingTime(std::chrono::milliseconds& current)
+    {
+      if (m_testContext.IsPlaybackMode())
+      {
+        current = 0ms;
+      }
+    }
+
+    // Util for tests to introduce delays
+    void TestDelay(std::chrono::milliseconds const& ms = 1s)
```

Comment:
Making the default be 0ms is a better default.  0ms on most platforms is a `yield` and only pauses if a waiting thread is ready for execution.

Reply:
We want to wait for one second by default. using 0 make sense only when test is running on playback so there's no network interaction